lib/parallel_tests/grouper.rb

```diff
@@ -28,7 +28,39 @@ def in_even_groups_by_size(items, num_groups, options = {})
           raise 'Number of isolated processes must be less than total the number of processes'
         end
 
-        if isolate_count >= 1
+        if options[:specify_groups]
+          specify_test_process_groups = options[:specify_groups].split('|')
+          all_specified_tests = specify_test_process_groups.map { |group| group.split(',') }.flatten
+          if specify_test_process_groups.count > num_groups
+            raise 'Number of processes separated by pipe must be less than or equal to the total number of processes'
+          end
+
+          specified_items_found, items = items.partition do |item, _size|
+            all_specified_tests.any? { |specified_spec| item == specified_spec }
+          end
+
+          specified_specs_not_found = all_specified_tests - specified_items_found.map(&:first)
+          if specified_specs_not_found.any?
+            raise "Could not find #{specified_specs_not_found} from --specify-groups in the main selected files & folders"
+          end
+
+          if specify_test_process_groups.count == num_groups && items.flatten.any?
```

Comment:
not sure why we'd need to flatten
```suggestion
          if specify_test_process_groups.count == num_groups && items.any?
```

Reply:
Since items is an array of arrays, just in case it shows up like `[[]]` instead of `[]` when empty, that was what I was thinking

But maybe when its empty it'd just be `[]` so no need to flatten?

Reply:
yeah, might be good as is